### PR TITLE
test: Add subcommand documentation verification

### LINKED ISSUE #1596: Enhance help sync test to verify subcommand documentation
## Summary

Follow-up to #1594. The current help sync test only checks if commands **exist** in help documentation, not if their subcommands/options are correctly documented.

## Problem

PR #1593 fixed outdated help documentation where:
- `tags` showed old commands (`add-tag`, `use-tag`) instead of new subcommand structure (`tags add`, `tags use`)
- `list` was missing new options (`--watch`, `--ready`, `--blocking`)

The test in #1595 wouldn't catch these issues because it only verifies command existence.

## Proposed Enhancement

Add verification that:
1. `tags` subcommands are documented (`tags add`, `tags use`, `tags remove`, `tags rename`, `tags copy`, `tags list`)
2. Key command options are documented (e.g., `list` shows `--watch`, `--ready`, `--blocking`)

## Dependencies

- Requires #1593 to be merged first (or branch from it)

## Related

- #1588 - Original help outdated issue
- #1593 - Fix for outdated help
- #1594 - Original test issue
- #1595 - Basic help sync test PR

### PR DESCRIPTION
**Pull request by bjcoombs**
## Summary

Enhances the help documentation sync test to verify that subcommand documentation stays in sync with CLI implementations.

## Changes

- Adds tests to verify `tags` subcommands use the new unified structure (`tags add`, `tags use`, etc.)
- Verifies deprecated standalone tag commands (`add-tag`, `use-tag`) are not documented
- Verifies `list` command options are documented (`--watch`, `--ready`, `--blocking`)
- Verifies `list` format options are documented (`--json`, `-f`, `-c`)

## Test Plan

- [x] All 10 tests pass locally
- [x] Tests correctly detect when subcommands are missing from help

## Dependencies

⚠️ **This PR depends on #1593** (help docs update) and should be merged after it.

This PR includes the help documentation updates from #1593 so it can verify against them. If #1593 changes, this PR will need to be rebased.

Closes #1596

## Summary by CodeRabbit

* **New Features**
  * Unified "tags" commands (add/use/remove/rename/copy) and added a new optional --from-branch flag for tags add.
  * Extended list command options and output formatting controls.

* **Documentation**
  * Updated CLI help text to reflect new tag command usages and remove deprecated standalone tag entries.

* **Tests**
  * Added comprehensive help-vs-CLI synchronization tests covering subcommands, tag mappings, list options, and extraction robustness.

---

> [!NOTE]
> **Low Risk**
> Test- and help-text-only changes; main risk is increased CI strictness causing failures if command/help strings drift or parsing heuristics miss edge cases.
> 
> **Overview**
> Adds a new `help-documentation-sync.spec.ts` unit test that parses legacy `commands.js`, modern `*.command.ts`, and `ui.js` help entries to fail CI when commands are missing from help or when help documents commands that no longer exist, with explicit allowlists/mappings for intentionally-undocumented and migrating commands.
> 
> Extends coverage to *subcommand correctness*: asserts help documents the unified `tags add/use/remove/rename/copy` structure, ensures deprecated standalone tag commands aren’t documented as primary entries, and verifies `list` help includes key option variants (watch/ready/blocking and format flags).
> 
> Updates the help text for `tags add` in `ui.js` to include the missing `--from-branch` argument, and adds a patch changeset describing the test/documentation improvements.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 37a3000db3426fd40726c48200651e9812f1f4f3. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by bjcoombs** [comment]
## Merge Order Note

This PR is branched from #1593 and includes its commits. Please merge in this order:

1. **#1593** first (help docs update)
2. **#1597** second (this PR - enhanced tests)

Alternatively, merging this PR alone will include #1593's changes, but merging #1593 first keeps the history cleaner.

File: apps/cli/tests/unit/help-documentation-sync.spec.ts [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+	'login', // Hamster login
+	'logout', // Hamster logout
+	'briefs', // Hamster briefs management
+	'context', // Hamster context management
+	'export', // Hamster export functionality
+	'export-tag', // Hamster export tag alias
+	'start', // Hamster workflow start
+	'loop', // Autonomous loop mode
+	'generate' // May be internal or aliased
+];
```
[the remark this answers is not in the document]
Comment (bjcoombs) [reply apps/cli/tests/unit/help-documentation-sync.spec.ts:140]: Fixed - removed `generate` from INTENTIONALLY_UNDOCUMENTED list.

**Review by Crunchyman-ralph — APPROVED** [review]
love this! I think it needs a changeset, don't you ?
